low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -20,49 +20,240 @@
  ****************************************************************************
 """
 
-import time
 import socket
-import copy
-import os
+import time
+from enum import Enum
+from queue import Queue
 
-from dbus import Interface, SystemBus, DBusException, PROPERTIES_IFACE
+from dbus import PROPERTIES_IFACE, DBusException, Interface, SystemBus
 from dbus.mainloop.glib import DBusGMainLoop
 from gi.repository import GLib
-
 from zope.interface import implementer
-from sensors.ISystem_monitor import ISystemMonitor
+
 from framework.base.internal_msgQ import InternalMsgQ
-from framework.base.module_thread import SensorThread
-from message_handlers.service_msg_handler import ServiceMsgHandler
-from framework.utils.service_logging import logger
+from framework.base.module_thread import SensorThread, ThreadException
 from framework.base.sspl_constants import DATA_PATH
-from framework.utils.conf_utils import ( GLOBAL_CONF, SSPL_CONF,
-    Conf, NODE_ID_KEY)
-from framework.utils.severity_reader import SeverityReader
-from framework.utils.mon_utils import get_alert_id
+from framework.utils.conf_utils import (SSPL_CONF, Conf)
 from framework.utils.iem import Iem
-from framework.base.module_thread import ThreadException
-from framework.utils.store_factory import file_store
+from framework.utils.mon_utils import get_alert_id
+from framework.utils.service_logging import logger
+from framework.utils.severity_reader import SeverityReader
+from framework.utils.store_factory import store
+from message_handlers.service_msg_handler import ServiceMsgHandler
+from sensors.ISystem_monitor import ISystemMonitor
+
+UNIT_IFACE = "org.freedesktop.systemd1.Unit"
+SERVICE_IFACE = "org.freedesktop.systemd1.Service"
+MANAGER_IFACE = 'org.freedesktop.systemd1.Manager'
+SYSTEMD_BUS = "org.freedesktop.systemd1"
+CACHE_PATH = f"/{DATA_PATH}/server/SERVICE_MONITOR/"
+
+inactive_services = set()
+properties_changed_not_subscribed_services = set()
+alerts = Queue()
+
+
+class FailedAlert(Enum):
+    alert_type = "fault"
+    description = "{} in {} state."
+    impact = "{} service is unavailable."
+    recommendation = "Try to restart the service"
+
+
+class InactiveAlert(Enum):
+    alert_type = "fault"
+    description = "{} in a {} state for more than {} seconds."
+    impact = "{} service is unavailable."
+    recommendation = "Try to restart the service"
+
+
+class ResolvedAlert(Enum):
+    alert_type = "fault_resolved"
+    description = "{} in {} state."
+    impact = "{} service is available now."
+    recommendation = ""
+
+
+class ActiveState:
+    failed = False
+
+    @staticmethod
+    def enter(service):
+        inactive_services.discard(service.name)
+
+
+class FailedState:
+    failed = True
+
+    @staticmethod
+    def enter(service):
+        inactive_services.discard(service.name)
```

Comment:
Discarding because if service is failed(can happen if failed alert is generated  or inactive alert is generated) , we don't need to check if service is inactive for 180 seconds or not.